server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -261,13 +288,19 @@ void becomeFollower(String method, DiscoveryNode leaderNode) {
             mode = Mode.FOLLOWER;
             joinAccumulator.close(mode);
             joinAccumulator = new JoinHelper.FollowerJoinAccumulator();
+            leaderChecker.setCurrentNodes(DiscoveryNodes.EMPTY_NODES);
         }
 
         lastKnownLeader = Optional.of(leaderNode);
         peerFinder.deactivate(leaderNode);
         closePrevotingAndElectionScheduler();
         cancelActivePublication();
         preVoteCollector.update(getPreVoteResponse(), leaderNode);
+
+        if (leaderCheckScheduler != null) {
+            leaderCheckScheduler.close();
+        }
+        leaderCheckScheduler = leaderChecker.startLeaderChecker(leaderNode);
```

Comment:
I'm confused. Does this mean we restart the leader checker on every incoming publication? We call becomeFollower on every incoming publication

Reply:
It did mean a new leader checker on each publication. I pushed be15266415f7d6d2e9ae586e219fe26400622fb0 not to do this. However I can't think of a good way to reliably assert that we don't do this: both ways have pretty much the right liveness properties, and my other two ideas are:

- check we don't send another leader check immediately after each publication (not robust)
- check for reference equality of the `leaderCheckScheduler` object before/after a second publication (don't fancy exposing this).

Reply:
> However I can't think of a good way to reliably assert that we don't do this
I don't have any good idea here. I'll keep thinking about this. Should not block this PR though.